d2-int-test/src/test/java/com/linkedin/d2/loadbalancer/strategies/TestLoadBalancerStrategy.java

```diff
@@ -0,0 +1,116 @@
+package com.linkedin.d2.loadbalancer.strategies;
+
+import com.linkedin.d2.balancer.strategies.framework.LatencyQPSCorrelation;
+import com.linkedin.d2.balancer.strategies.framework.LoadBalancerStrategyTestRunner;
+import com.linkedin.d2.balancer.strategies.framework.LoadBalancerStrategyTestRunnerBuilder;
+import com.linkedin.d2.loadBalancerStrategyType;
+import java.net.URI;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import org.testng.annotations.Test;
+
+import static com.linkedin.d2.balancer.properties.PropertyKeys.*;
+import static org.testng.Assert.*;
+
+
+/**
+ * Integration tests for all the load balancer strategies
+ */
+public class TestLoadBalancerStrategy {
```

Comment:
This framework is a great start. In addition to these strategy specific tests, can we define test scenarios generically in such a way that any strategy can be plugged into it? I'm imagining something comparable to a DataProvider in unit tests.

We can define several scenarios which we will consider our "standards" that any strategy should pass. For example:
1. A cluster of 5 hosts has 1 extremely unhealthy. We can assert that in 5(?) intervals, that host should be fully degraded by any strategy using its default configs. 
2. Continuing the previous scenario, a fully degraded host is now perfectly healthy. We can assert that in 5 intervals the host is now fully recovered by any strategy using its default configs.
3. etc, etc

Reply:
Added a few default test scenarios